src/util/listQueryBuilder.js

```diff
@@ -529,57 +529,64 @@ async function performHistoryQuery ({
   const interval = intervals[groupBy]
   const view = groupByViews[groupBy]
 
-  let allResults
-
-  if (useOnlyNonTimeRestrictedFilters && view) {
-    // `AT TIME ZONE 'UTC'` statement is important to add to have the type 'timestamptz'
-    // otherwise the type 'timestamp' is returned and the date won't be UTC
-    let select
-    if (secondaryFilter) {
-      select = knex.raw('?? AT TIME ZONE \'UTC\' as ??, sum(count)::REAL as count', [groupBy, groupBy])
+  const useContinuousAggregate = useOnlyNonTimeRestrictedFilters && view
+
+  const select = (() => {
+    const selectParams = []
+    let selectQuery = ''
+
+    if (useContinuousAggregate) {
+      // Please consult continuous aggregate definition in `migrations/util/timescaleDB.js`
+      // it already embeds `time_bucket` function, thus no need to specify it via request
+      // only reference it via `groupBy` value (hour, day or month)
+      selectQuery += '??'
+      selectParams.push(groupBy)
     } else {
-      select = knex.raw('?? AT TIME ZONE \'UTC\' as ??, count', [groupBy, groupBy])
+      selectQuery += `public.time_bucket(INTERVAL '${interval}', ??)`
+      selectParams.push(timeColumn)
     }
 
-    queryBuilder = queryBuilder
-      .select(select)
-      .from(knex.raw('??.??', [schema, view]))
+    // `AT TIME ZONE 'UTC'` statement is important to add to have the type 'timestamptz'
+    // otherwise the type 'timestamp' is returned and the date won't be UTC
+    selectQuery += ' AT TIME ZONE \'UTC\' as ??'
+    selectParams.push(groupBy)
 
-    if (filters) {
-      const newFilters = _.pick(filters, [timeFilter, secondaryFilter])
-      newFilters[timeFilter].dbField = groupBy
-      const transformedValues = getTransformedValues(newFilters)
-      addFiltersToQueryBuilder(queryBuilder, newFilters, transformedValues)
+    selectQuery += ', '
 
-      if (secondaryFilter) queryBuilder.groupBy(groupBy)
+    if (useContinuousAggregate) {
+      if (secondaryFilter) selectQuery += 'sum(count)::REAL as count'
+      else selectQuery += 'count'
+    } else {
+      selectQuery += 'count(*)'
```

Comment:
This is the most confusing part in your string concatenation.

It’s unclear why use `count` in one case and `count(*)` in the other. This relates to use of `knex.count()` you removed below.

Reply:
A comment was needed to be clear.

When using the continuous aggregate, `count` is a column value which is pre-aggregated, so we can use it directly.
In the other case, `count(*)` (equivalent to `knex.count()`) must be performed on the fly.

I'll add a comment for that.